src/includes/blocks/base_block.h

```diff
@@ -19,9 +19,10 @@ typedef struct Block{
 
 #endif
 
-char *ser_blockheader(char *dest, BlockHeader *block_header);
-char *ser_blockheader_alloc(BlockHeader *block_header);
+unsigned char *ser_blockheader(unsigned char *dest, BlockHeader *block_header);
+unsigned char *ser_blockheader_alloc(BlockHeader *block_header);
+void hash_blockheader(BlockHeader *header, unsigned char *buf);
```

Comment:
Anything that takes a buffer that is being written into should have a block comment above it in header so we have tooltips on hover. as well as know the appropriate size of the buffer needed. also we should start writing block comments for functionality at this point I think

Reply:
I think this applies to a lot of the functions written in this pr